views/default/navigation/pagination.php

```diff
@@ -37,6 +40,21 @@
 	$base_url = current_page_url();
 }
 
+if ($url_fragment && !preg_match('~#.~', $base_url)) {
+	// we'll sometimes add fragment
+	$get_href = function ($offset) use ($base_url, $offset_key, $url_fragment) {
+		$link = elgg_http_add_url_query_elements($base_url, array($offset_key => $offset));
+		if ($offset) {
+			$link .= "#$url_fragment";
+		}
```

Comment:
wouldn't this prevent you from going from offset=10 to offset=0 (so back to the first page) not jump to the comments section